notebooks/Notebook3.md

```diff
@@ -18,17 +20,31 @@ exciting milestones, preliminary results, etc. Feel free to include images
 code, and any other resources that you think will help clearly convey your
 design process.
 
+My main goal for this week was to get sbt working on Codespaces. However, I got sick and couldn't make it to class or any of the tutoring hours this week and still couldn't figue out how to set it up on my own. Thankfully, I figured out a way to code through a work sheet which automatically displays the output for each line without having to compile. Next week it will still be important to get sbt working. 
+
+The code I wrote handles how to input different foods, meals, and weights. These are currently stored in csv files so that your data can be saved over time. I created four different csv files: food, meal, weight, and foods eaten. "food.csv" comtains the name of each food, the amount of calories, and any tags that are added to it. "meal.csv" contains the name of each meal and which food it contains. "weight.csv" contains the weight of each weigh-in and the date and time it occured. "foods_eaten.csv" contains the name of each food and the date and time it was eaten. 
+
+One thing I am considering is finding a way to just save variables over time instead of using a csv file. I am not sure if there is a way to store variable on disk, but I will ask in class because that would be way more convenient than using csv files. 
+
+Right now, a new food is automatically saved when an instance of the class Food is created. I still need to add what happens when a food is updated. I am thinking of making the other parameters optional, so if the food already exists then you can just enter the string to log it, but this wouldn't be necessary if I can just use saved variables instead of csv files. 
+
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+I need to figure out how I am going to store all my data and then start working on the functions for creating graphs. I also need to add a function for tracking weight, but that will be easy. 
+
 **What questions do you have for your critique partners? How can they best help
 you?**
 
+I have added tags to the food class, but would it be helpful to have tags for meals as well? Do you know of any way to save variables to disk, so that when you return to the language, they will still be saved? 
```

Comment:
Maybe your code could create a txt file or csv file and store it in that, and then use the file when needed? 

I really don't know though, I am sure there are better ways.

Reply:
A technical term related to this is [serialization](https://en.wikipedia.org/wiki/Serialization). You can do it really easily in Python with [`pickle`](https://docs.python.org/3/library/pickle.html); in Java it's more involved. A Google search for [`Scala serialization`](https://www.google.com/search?q=Scala+serialization) reveals some stuff you could try, but I'm not in a great position to evaluate whether that'd be easier or harder than just using a CSV.